test/index.js

```diff
@@ -36,12 +36,48 @@ describe('KeyringController', function () {
   })
 
   afterEach(function () {
-    // Cleanup mocks
     sandbox.restore()
   })
 
+  describe('setLocked', function () {
+
+    it('setLocked correctly sets lock state', async function () {
+
+      assert.notDeepEqual(
+        keyringController.keyrings, [],
+        'keyrings should not be empty',
+      )
+
+      await keyringController.setLocked()
+
+      assert.equal(
+        keyringController.password, null,
+        'password should be null',
+      )
+      assert.equal(
+        keyringController.memStore.getState().isUnlocked, false,
+        'isUnlocked should be false',
+      )
+      assert.deepEqual(
+        keyringController.keyrings, [],
+        'keyrings should be empty',
+      )
+    })
+
+    it('emits "lock" event', async function () {
+
+      let called = false
+      keyringController.on('lock', () => {
```

Comment:
Can we pass a spy as the callback? That would allow us to avoid keeping this state ourselves.

Reply:
Done for both `lock` and `unlock` event tests in db251f9